StyleCop.Analyzers/StyleCop.Analyzers/Helpers/NamedTypeHelpers.cs

```diff
@@ -76,6 +77,15 @@ internal static bool IsContainedInNativeMethodsClass(SyntaxNode syntax)
             return false;
         }
 
+        internal static string GetNameOrIdentifier(MemberDeclarationSyntax member)
+        {
+            string name = null;
+            name = name ?? (member as TypeDeclarationSyntax)?.Identifier.Text;
+            name = name ?? (member as EnumDeclarationSyntax)?.Identifier.Text;
+            name = name ?? (member as DelegateDeclarationSyntax)?.Identifier.Text;
+            return name;
```

Comment:
❗️ This code can be done more efficiently with a `switch` statement on `member.Kind()`